src/pip/_internal/format_control.py

```diff
@@ -0,0 +1,64 @@
+from pip._vendor.packaging.utils import canonicalize_name
+
+
+class FormatControl(object):
+    """This class has two fields, no_binary and only_binary.
+    If a field is falsy, it isn't set. If it is {':all:'}, it should match all
+    packages except those listed in the other field. Only one field can be set
+    to {':all:'} at a time. The rest of the time exact package name matches
+    are listed, with any given package only showing up in one field at a time.
+    """
+    def __init__(self, no_binary, only_binary):
+        self.no_binary = no_binary
+        self.only_binary = only_binary
+
+    def _handle_no_binary(self, option, opt_str, value, parser):
+        existing = getattr(parser.values, option.dest)
+        self.fmt_ctl_handle_mutual_exclude(
+            value, existing.no_binary, existing.only_binary,
+        )
+
+    def _handle_only_binary(self, option, opt_str, value, parser):
```

Comment:
This should probably be renamed to be `handle_cli_only_binary`.

Reply:
Actually, this and it's friend function can simply stay as is in `cmdoptions`.

Reply:
@pradyunsg I think moving `handle_cli_only_binary` why do you want to keep `handle_cli_only_binary` and `handle_cli_no_binary` to `cmdoptions` as is? Just asking. Because I think moving them to a separate class is better and also they are handling things related to format control and should be in the same class.

Reply:
Oh, that's what I was suggesting: all three handle_cli_only_binary, handle_cli_no_binary, _get_format_control moving back into cmdoptions, with one change -- _get_format_control actually gets used by the other two functions.

IMO -- they're basically for handling the CLI part of things (and specific to optparse); so they should live within the cli submodule.

Reply:
sounds good. Well, I will make the changes soon. 😄

Reply:
Awesome. :)